src/wing/CCPACSWingSparSegment.cpp

```diff
@@ -508,34 +511,52 @@ void CCPACSWingSparSegment::BuildSparCapsGeometry(SparCapsCache& cache) const
 // Builds the cutting geometry for the spar as well as the midplane line
 gp_Pnt CCPACSWingSparSegment::GetMidplanePoint(const std::string& positionUID) const
 {
-    gp_Pnt midplanePoint;
-    CCPACSWingSparPosition& sparPosition               = sparsNode.GetSparPositions().GetSparPosition(positionUID);
-    CTiglWingStructureReference wsr(*sparsNode.GetParent());
+    CCPACSWingSparPosition& sparPosition = sparsNode.GetSparPositions().GetSparPosition(positionUID);
 
-    if (sparPosition.GetInputType() == CCPACSWingSparPosition::ElementUID) {
-            const CCPACSWingComponentSegment& componentSegment = wsr.GetWingComponentSegment();
-        midplanePoint = getSectionElementChordlinePoint(componentSegment, sparPosition.GetElementUID(), sparPosition.GetXsi());
-    }
-    else if (sparPosition.GetInputType() == CCPACSWingSparPosition::Eta) {
-        midplanePoint = wsr.GetPoint(sparPosition.GetEta(), sparPosition.GetXsi(), WING_COORDINATE_SYSTEM);
-    }
-    else {
-        throw CTiglError("Unkwnonw SparPosition InputType found in CCPACSWingSparSegment::GetMidplanePoint");
-    }
-    return midplanePoint;
+    return GetSparMidplanePoint(sparPosition, *sparsNode.GetParent());
 }
 
 gp_Vec CCPACSWingSparSegment::GetUpVector(const std::string& positionUID, gp_Pnt midplanePnt) const
 {
     const CCPACSWingSparPosition& position = sparsNode.GetSparPositions().GetSparPosition(positionUID);
     const CTiglWingStructureReference wsr(*sparsNode.GetParent());
 
-    if (position.GetInputType() == CCPACSWingSparPosition::ElementUID) {
+    if ((position.isOnSectionElement() || position.isOnRib()) && wsr.GetType() == CTiglWingStructureReference::ComponentSegmentType) {
         // get componentSegment required for getting chordline points of sections
         const CCPACSWingComponentSegment& componentSegment = wsr.GetWingComponentSegment();
 
+        TopoDS_Shape sectionFace;
+        if (position.isOnSectionElement()) {
+            std::string sectionElemUID;
```

Comment:
Why don't you use `WingSparPosGetElementUID` here?
```suggestion
            const std::string sectionElemUID = WingSparPosGetElementUID(position);
```

Reply:
Good point. I suppose, I developed the function ```WingSparPosGetElementUID``` later and forgot to use it then at this position.

Reply:
Changed!